torch/__init__.py

```diff
@@ -426,11 +426,41 @@ def use_deterministic_algorithms(d):
     Note that deterministic operations tend to have worse performance than
     non-deterministic operations.
 
+    .. note::
+
+        :func:`use_deterministic_algorithms` is not guaranteed to prevent
+        nondeterministic behavior that is caused by an operation performing
+        multiple writes of different data to a single memory location. The
+        order of writes to a single location is not guaranteed, which can cause
+        nondeterminism. This could occur when an output tensor contains internal
+        memory overlaps or if an indexed operation is given duplicate indices.
+        Below are several examples.
+
+    .. code-block:: python
+
+        >>> torch.use_deterministic_algorithms(True)
+
+        # Determinism is not guaranteed for an in-place operation on a tensor
+        # with internal memory overlap
+        >>> a = torch.tensor(1.).as_strided((3,), (0,))
+        >>> a.index_put_((torch.tensor([0, 1, 2]),), torch.tensor([0., 1., 2.]))
+        tensor([2., 2., 2.])
+        >>> a = torch.tensor(1., device='cuda').as_strided((3,), (0,))
+        >>> a.index_put_((torch.tensor([0, 1, 2]).cuda(),), torch.tensor([0., 1., 2.]).cuda())
+        tensor([0., 0., 0.], device='cuda:0')
+
+        # Determinism is not guaranteed for duplicate indices
+        >>> a = torch.zeros(3)
+        >>> a.index_put_((torch.tensor([1, 1, 1]),), torch.tensor([1., 2., 3.]))
+        tensor([0., 3., 0.])
+        >>> a.cuda().index_put_((torch.tensor([1, 1, 1]).cuda(),), torch.tensor([1., 2., 3.]).cuda())
+        tensor([0., 1., 0.], device='cuda:0')
+
     Args:
-        d (:class:`bool`): If True, force operations to be deterministic.
-                           If False, allow non-deterministic operations.
+        mode (:class:`bool`): If True, force operations to be deterministic.
```

Comment:
'force' -> 'forces'
'allow' -> 'allows'

Conceptually it's the mode taking action, so the mode "forces" the operations. 

What about elaborating slightly in the first case?

"If True, makes potentially nondeterministic operations switch to a deterministic algorithm or throw a runtime error.